src/newViews/Explore/LoginRegisterModal/LoginForm/index.tsx

```diff
@@ -0,0 +1,318 @@
+            authenticate();
+        },
+        onFailure: ({ errorCode, value: errorValue }) => {
+            if (errorCode === 4004) {
+                onErrorSet({
+                    fields: {
+                        ...errorValue.faramErrors,
+                    },
+                    $internal: [
+                        captchaRequired
+                            ? _ts('explore.login', 'retryRecaptcha')
+                            : _ts('explore.login', 'enterRecaptcha'),
+                    ],
+                });
+                setCaptchaRequired(true);
+            } else {
+                onErrorSet({
+                    fields: { ...errorValue.faramErrors },
+                    $internal: errorValue.faramErrors.$internal,
+                });
+            }
+        },
+        schemaName: 'tokenGetResponse',
+    });
+
+    const {
+        trigger: hidLoginTrigger,
+    } = useLazyRequest<LoginResponse, HidParams>({
+        url: 'server://token/',
+        method: 'POST',
+        body: ctx => ctx,
+        onSuccess: ({ refresh, access }) => {
+            login({ refresh, access });
+            startSiloTasks(() => console.log('Silo tasks started'));
+            authenticate();
+        },
+        onFailure: ({ errorCode, value: errorValue }) => {
+            if (errorCode === 4004) {
+                onErrorSet({
+                    fields: {
+                        ...errorValue.faramErrors,
+                    },
+                    $internal: [
+                        captchaRequired
+                            ? _ts('explore.login', 'retryRecaptcha')
+                            : _ts('explore.login', 'enterRecaptcha'),
+                    ],
+                });
+                setCaptchaRequired(true);
+            } else {
+                onErrorSet({
+                    fields: { ...errorValue.faramErrors },
+                    $internal: errorValue.faramErrors.$internal,
+                });
+            }
+        },
+        schemaName: 'tokenGetResponse',
+    });
+
+    // FIXME: We need to change the redirect URL in hid configurations later on
+    const checkParamsFromHid = useCallback(() => {
+        // Get params from the current url
+        // NOTE: hid provides query as hash
+        // eslint-disable-next-line camelcase
+        const query = parseUrlParams(location?.hash?.replace('#', '')) as { access_token?: string };
+        // Login User with HID access_token
+        if (query.access_token) {
+            const hidQuery = query as HidQuery;
+            const params = {
+                accessToken: hidQuery.access_token,
+                expiresIn: hidQuery.expires_in,
+                state: hidQuery.state,
+                tokenType: hidQuery.token_type,
+            };
+            hidLoginTrigger(params);
+        }
+    }, [hidLoginTrigger, location]);
+
+    useEffect(() => {
+        checkParamsFromHid();
+    }, [checkParamsFromHid]);
```

Comment:
This looks a bit weird. Why define a separate named function checkParamsFromHid?